fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java

```diff
@@ -139,9 +139,11 @@ public boolean registerConnection(ConnectContext ctx) {
         if (currentConn >= currentMaxConn) {
             return false;
         }
-        numberConnection.incrementAndGet();
-        connCountByUser.get(ctx.getQualifiedUser()).incrementAndGet();
-        connectionMap.put((long) ctx.getConnectionId(), ctx);
+        ConnectContext existingConnectionContext = connectionMap.putIfAbsent((long) ctx.getConnectionId(), ctx);
+        if (existingConnectionContext == null) {
```

Comment:
Do you know when this will be null?

Reply:
When a connection is established,registerConnection has two callers:
1 handleEvent->registerConnection
2 handleEvent->connectScheduler.submit->executor.submit->LoopHandler.run->registerConnection
The first caller to registerConnection enters the existingConnectionContext == null block of code。

Reply:
These two calls will not occur at the same time. One is for NIO, anther is for Block IO.